src/charm.py

```diff
@@ -45,7 +45,7 @@ def _resource_spec_from_config(self) -> ResourceRequirements:
         return adjust_resource_requirements(resource_limit, None)
 
     def _on_resource_patch_failed(self, event: K8sResourcePatchFailedEvent):
-        self.unit.status = BlockedStatus(event.message)
+        self.unit.status = BlockedStatus(str(event.message))
```

Comment:
We shouldn't need a `str` here. We should probably fix the type of `message`.

Reply:
This charm is not even published, so I doubt we would notice the error from this charm; regardless it's a pretty small change, so sure!